src/main/java/picard/illumina/IlluminaBasecallsToSam.java

```diff
@@ -305,17 +305,18 @@ private void initialize() {
             adapters.add(new CustomAdapterPair(FIVE_PRIME_ADAPTER, THREE_PRIME_ADAPTER));
         }
 
+        final boolean demultiplex = !readStructure.sampleBarcodes.isEmpty();
```

Comment:
It looks like this expression is used elsewhere, maybe it should be a method on `ReadStructure`, e.g. `hasSampleBarcode()`?